sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,440 @@
+      resource.Filter = {
+        SqlExpression: "1=1",
+        CompatibilityLevel: 20
+      };
+      resource.Filter[Constants.XML_METADATA_MARKER] = {
+        "p4:type": "SqlFilter",
+        "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+      };
+    } else {
+      let isSqlFilter: boolean = false;
+
+      const givenFilter: any = rule.filter as any;
+
+      if (givenFilter.sqlExpression != undefined) {
+        isSqlFilter = true;
+      } else if (
+        !givenFilter.correlationId ||
+        !givenFilter.label ||
+        !givenFilter.to ||
+        !givenFilter.replyTo ||
+        !givenFilter.replyToSessionId ||
+        !givenFilter.contentType ||
+        !givenFilter.sessionId ||
+        !givenFilter.messageId ||
+        !givenFilter.userProperties
+      ) {
+        isSqlFilter = false;
+      }
+
+      if (isSqlFilter) {
+        const sqlFilter: SqlFilter = rule.filter as SqlFilter;
+        resource.Filter = {
+          SqlExpression: sqlFilter.sqlExpression,
+          Parameters: getRawSqlParameters(sqlFilter.sqlParameters),
+          CompatibilityLevel: 20,
+          RequiresPreprocessing: getStringOrUndefined(sqlFilter.requiresPreprocessing)
+        };
+        resource.Filter[Constants.XML_METADATA_MARKER] = {
+          "p4:type": "SqlFilter",
+          "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+        };
+      } else {
+        const correlationFilter: CorrelationFilter = rule.filter as CorrelationFilter;
+
+        resource.Filter = {
+          CorrelationId: correlationFilter.correlationId,
+          Label: correlationFilter.label,
+          To: correlationFilter.to,
+          ReplyTo: correlationFilter.replyTo,
+          ReplyToSessionId: correlationFilter.replyToSessionId,
+          ContentType: correlationFilter.contentType,
+          SessionId: correlationFilter.sessionId,
+          MessageId: correlationFilter.messageId,
+          Properties: correlationFilter.userProperties
+        };
+        resource.Filter[Constants.XML_METADATA_MARKER] = {
+          "p4:type": "CorrelationFilter",
+          "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+        };
+      }
+    }
+
+    if (rule == undefined || rule.action == undefined) {
+      // Defaults to creating an empty rule action instance if none specified
+      resource.Action = {};
+      resource.Action[Constants.XML_METADATA_MARKER] = {
+        "p4:type": "EmptyRuleAction",
+        "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+      };
+    } else {
+      resource.Action = {
+        SqlExpression: rule.action.sqlExpression,
+        Parameters: getRawSqlParameters(rule.action.sqlParameters),
+        CompatibilityLevel: 20,
+        RequiresPreprocessing: getStringOrUndefined(rule.action.requiresPreprocessing)
+      };
+      resource.Action[Constants.XML_METADATA_MARKER] = {
+        "p4:type": "SqlRuleAction",
+        "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
```

Comment:
Should this string be added to a constant?

Reply:
Doesn't seem necessary, we had a constant earlier and decided to inline them since a couple of such strings exist